### PR TITLE
feat: Early vault exit improvements

### PR DESCRIPTION
**Pull request by marshall2112**
# Description

- Users can edit the amount in case of early exit (previously it was only editable in a normal claim scenario)
- Timing page, fixed a bug where the `-` was clickable when it should not have been
- I guess the typechain files got updated somewhere along the way

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

**Comment by marshall2112** [comment]
Not sure what happened with the vercel deploy, looking at it. But meanwhile, it's here https://dapp-q5nwjk6pl-templedao.vercel.app/

**Comment by marshall2112** [comment]
Deploy preview for _dapp_ ready!

✅ Preview
https://dapp-oe7w6rlna-templedao.vercel.app
https://pr-599-preview.templedao.link

Built with commit f3446a424c56aa399b11de0ebda55bd093821b5a.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

File: apps/dapp/src/types/typechain/factories/GenericZap__factory.ts
Comment (frontier159) [inline apps/dapp/src/types/typechain/factories/GenericZap__factory.ts:1144]: interesting that the typechain files were updated?

File: apps/dapp/src/hooks/core/use-withdraw-from-vault.tsx
Comment (frontier159) [inline apps/dapp/src/hooks/core/use-withdraw-from-vault.tsx:28]: Out of interest - does ethers get the gas estimate wrong here, is that why you need to specify `gasLimit`?

Comment (marshall2112) [reply apps/dapp/src/hooks/core/use-withdraw-from-vault.tsx:28]: Basically yes, that's what was happening when I was manually testing.

**Review by nichosystem — APPROVED** [review]

File: apps/dapp/src/types/typechain/factories/GenericZap__factory.ts
Comment (frontier159) [reply apps/dapp/src/types/typechain/factories/GenericZap__factory.ts:1144]: It may only be the embedded metadata at the end of the bytecode that changed... Perhaps different yarn/hardhat config?

For now to unblock this, I'd remove from the pr as we know what's there works. I can check what might have changed

**Review by nichosystem — APPROVED** [review]

**Comment by frontier159** [comment]
@nichosystem hmm no requirement to resolve all comments before merging? It's a good feature to have switched on imo

**Pull request merged**
